ts/api/api.ts

```diff
@@ -185,109 +205,136 @@ class OpenYoloApiImpl implements OpenYoloApi {
-          // Ignore errors.
-          return false;
-        });
+    let wrapBrowser = false;
+    try {
+      wrapBrowser = await request.dispatch(undefined, timeoutRacer);
+    } catch (e) {
+      // Only rethrow a timeout error.
+      timeoutRacer.rethrowTimeoutError(e);
+      // Default to false if request fails.
+    }
 
-    return new OpenYoloApiImpl(
-        frameManager, channel, wrapBrowser, !!areTimeoutsDisabled);
+    if (wrapBrowser) {
+      return new OpenYoloBrowserApiImpl();
+    }
+    return new OpenYoloApiImpl(frameManager, channel);
+  } catch (e) {
+    timeoutRacer.handleTimeoutError(e);
+    // Convert the timeout error.
+    throw OpenYoloError.requestTimeout();
+  }
+}
+
+/**
+ * OpenYolo implementation using the browser navigator.credentials.
+ * TODO: Implement.
+ */
+class OpenYoloBrowserApiImpl implements OpenYoloWithTimeoutApi {
+  async retrieve(options: CredentialRequestOptions, timeoutRacer: TimeoutRacer):
+      Promise<Credential|null> {
+    return Promise.reject('not implemented');
+  }
+
+  async hintsAvailable(
+      options: CredentialRequestOptions,
+      timeoutRacer: TimeoutRacer): Promise<boolean> {
+    return Promise.reject('not implemented');
+  }
+
+  async hint(options: CredentialRequestOptions, timeoutRacer: TimeoutRacer):
+      Promise<Credential|null> {
+    return Promise.reject('not implemented');
+  }
+
+  async disableAutoSignIn(timeoutRacer: TimeoutRacer): Promise<void> {
+    return Promise.reject('not implemented');
   }
 
+  async save(credential: Credential, timeoutRacer: TimeoutRacer):
+      Promise<void> {
+    return Promise.reject('not implemented');
+  }
+
+  async proxyLogin(credential: Credential, timeoutRacer: TimeoutRacer):
+      Promise<ProxyLoginResponse> {
+    return Promise.reject('not implemented');
+  }
+
+  async cancelLastOperation(timeoutRacer: TimeoutRacer): Promise<void> {
+    return Promise.reject('not implemented');
+  }
+}
+
+/**
+ * Provides access to the user's preferred credential provider, in order to
+ * retrieve credentials.
+ */
+class OpenYoloApiImpl implements OpenYoloWithTimeoutApi {
   private disposed: boolean = false;
 
   constructor(
       private frameManager: ProviderFrameElement,
-      private channel: SecureChannel,
-      private wrapBrowser: boolean,
-      private areTimeoutsDisabled: boolean) {}
+      private channel: SecureChannel) {}
 
-  async hintsAvailable(options: CredentialHintOptions): Promise<boolean> {
+  async hintsAvailable(
+      options: CredentialHintOptions,
+      timeoutRacer: TimeoutRacer): Promise<boolean> {
     this.checkNotDisposed();
     const request = new HintAvailableRequest(this.frameManager, this.channel);
-    const timeoutMs = this.areTimeoutsDisabled ?
-        undefined :
-        DEFAULT_TIMEOUTS.hintAvailableRequest;
-    return request.dispatch(options, timeoutMs).catch((error) => {
-      // Ignore errors.
+    try {
+      return await timeoutRacer.race(request.dispatch(options));
```

Comment:
Oversight from my part, good catch (it actually doesn't matter here as it's UI less but it's better to be consistent).